app/models/concerns/dirty_associations.rb

```diff
@@ -14,16 +14,36 @@
 module DirtyAssociations
   extend ActiveSupport::Concern
 
+  included do
+    after_save :reset_dirty_cache_state
```

Comment:
Note to myself:

This may be causing the audit test failures, audit is created `after_create` based on what is found in dirty association cache, see https://github.com/collectiveidea/audited/blob/master/lib/audited/auditor.rb#L73

though rails documentation says after_create is executed before after_save. Audited gem defines after_audit filter that we could use if this is the case. But I suppose we'd need to do it only for objects, that are audited. The rest shsould still use after_save.

Running the test locally, it passes.

Reply:
Looks like it just needed a rebase!